cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -111,12 +115,18 @@ public void setUp() throws Exception {
       doClear(configuredNamespace, false);
     }
     registeredNamespaces.put(configuredNamespace, deleteUponTeardown);
+    applicationManagers.clear();
     LOG.info("Completed setUp.");
   }
 
   @After
   public void tearDown() throws Exception {
     LOG.info("Beginning tearDown.");
+
+    // Stop any running programs created in @Test
+    for (ApplicationManager manager: applicationManagers) {
```

Comment:
"This would require a much bigger change - that endpoint returns only RUNNING or STOPPED"
The possible status in the run record can be mapped to one of these, to maintain the existing API.